# Autogenerated tests for SDAM error handling, see generate-error-tests.py
description: Post-4.2 InterruptedAtShutdown error
uri: mongodb://a/?replicaSet=rs
phases:
- description: Primary A is discovered
  responses:
  - - a:27017
    - ok: 1
      ismaster: true
      hosts:
      - a:27017
      setName: rs
      minWireVersion: 0
      maxWireVersion: 8
  outcome: &outcome
    servers:
      a:27017:
        type: RSPrimary
        setName: rs
        topologyVersion: null
        pool:
          generation: 0
    topologyType: ReplicaSetWithPrimary
    logicalSessionTimeoutMinutes: null
    setName: rs

- description: Post-4.2 InterruptedAtShutdown error marks server Unknown
  applicationErrors:
  - address: a:27017
    when: afterHandshakeCompletes
    maxWireVersion: 8
    type: command
    response:
      ok: 0
      errmsg: InterruptedAtShutdown
      code: 11600
  outcome:
    servers:
      a:27017:
        type: Unknown
        topologyVersion: null
        pool:
          generation: 1
    topologyType: ReplicaSetNoPrimary
    logicalSessionTimeoutMinutes: null
    setName: rs
